refinery_macros/src/lib.rs

```diff
@@ -116,25 +52,35 @@ pub fn embed_migrations(input: TokenStream) -> TokenStream {
         crate_root().join(location.value())
     };
 
-    let migration_files =
-        find_migration_files(location, MigrationType::Sql).expect("error getting migration files");
+    let migration_files = find_migration_files(location).expect("error getting migration files");
 
+    let mut migrations_mods = Vec::new();
     let mut _migrations = Vec::new();
-    for path in migration_files {
-        //safe to call unwrap as find_migration_filenames returns canonical paths
-        let filename = path
+
+    for migration in migration_files {
+        // safe to call unwrap as find_migration_filenames returns canonical paths
+        let filename = migration
             .file_stem()
             .and_then(|file| file.to_os_string().into_string().ok())
             .unwrap();
-        let path = path.display().to_string();
-        _migrations.push(quote! {
-            (#filename, include_str!(#path).to_string())
-        });
+        let path = migration.display().to_string();
+        let extension = migration.extension().unwrap();
+
+        if extension == "sql" {
+            _migrations.push(quote! {(#filename, include_str!(#path).to_string())});
+        } else if extension == "rs" {
```

Comment:
we now seem to have an issue where changes in the rust migrations don't trigger recompilation, what do you think of using a solution akin to https://github.com/djc/askama/pull/199 to address it?

Reply:
I didn't get this one. How would they trigger recompilation? You mean within `cargo watch`?

Reply:
no no. So with the `sql` migrations as we use `include_str!` everytime the sql files change, it triggers recompilation. Whereas with the `rs` migrations, the macro code itself doesn't change, so I was suggesting we use something like:

```Rust
//refinery_macros/src/lib.rs
        match extension 
            "sql"  => {
                _migrations.push(quote! {(#filename, include_str!(#path).to_string())});
              }
             "rs" => {
                 let rs_content = fs::read_to_string(&path)
                    .unwrap()
                    .parse::<TokenStream2>()
                    .unwrap();
                 let ident = Ident::new(&filename, Span2::call_site());
                 let mig_mod = quote! {pub mod #ident {
                     #rs_content
                     //also include the file as str so we recompile if it changes
                     const _rerun_if_changed: &'static str = include_str!(#path);
                 }};
                 _migrations.push(quote! {(#filename, #ident::migration())});
                 migrations_mods.push(mig_mod);
        }
 what do you think?

Reply:
Ah, yes. Now I understand.
And also like the workaround you've provided :D